Sources/ArcGISToolkit/FloorFilter/SiteFacilityPromptViewController.swift

```diff
@@ -0,0 +1,272 @@
+    }
+    
+    public func searchBarTextDidBeginEditing(_ searchBar: UISearchBar) {
+        isSearchActive = true
+        self.siteFacilitySearchBar.showsCancelButton = true
+    }
+    
+    public func searchBarSearchButtonClicked(_ searchBar: UISearchBar) {
+        isSearchActive = false
+        self.siteFacilitySearchBar.resignFirstResponder()
+        self.siteFacilitySearchBar.endEditing(true)
+    }
+    
+    public func searchBarCancelButtonClicked(_ searchBar: UISearchBar) {
+        isSearchActive = false
+        self.siteFacilitySearchBar.text = ""
+        self.siteFacilitySearchBar.resignFirstResponder()
+        self.siteFacilitySearchBar.endEditing(true)
+    }
+    
+    public func searchBarTextDidEndEditing(_ searchBar: UISearchBar) {
+        isSearchActive = false
+        self.siteFacilitySearchBar.resignFirstResponder()
+        self.siteFacilitySearchBar.endEditing(true)
+    }
+    
+    private func resetSearchFilteredResults() {
+        filteredSearchSites = viewModel.sites
+        filteredSearchFacilities = viewModel.facilities
+    }
+    
+    private func dismissSearchBar() {
+        self.isSearchActive = false
+        self.siteFacilitySearchBar.text = ""
+        self.siteFacilitySearchBar.resignFirstResponder()
+        self.siteFacilitySearchBar.endEditing(true)
+        self.siteFacilitySearchBar.showsCancelButton = false
+    }
+}
+
+/// Extension for the Sites and Facilities Table View
+extension SiteFacilityPromptViewController: UITableViewDataSource, UITableViewDelegate {
+    private func initializeSiteFacilityTableView() {
+        siteFacilityTableView.delegate = self
+        siteFacilityTableView.dataSource = self
+        siteFacilityTableView.separatorStyle = .none
+        siteFacilityTableView.register(UINib(nibName: "FloorFilterSiteFacilityCell", bundle: .module), forCellReuseIdentifier: "FloorFilterSiteFacilityCell")
+    }
+    
+    public func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        updatePromptTitle()
+        return isShowingFacilities ? getFilteredFacilities().count : getFilteredSites().count
+    }
+    
+    public func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        if let cell = siteFacilityTableView.dequeueReusableCell(withIdentifier: "FloorFilterSiteFacilityCell", for: indexPath) as? SiteFacilityTableViewCell {
+            let sites = getFilteredSites()
+            let facilities = getFilteredFacilities()
+        
+            // If there are no sites in the map, then directly show the facilities list
+            if (sites.isEmpty) {
+                isShowingFacilities = true
+            }
+            
+            cell.siteFacilityDotImg.isHidden = true
+            cell.siteFacilityNameLabel?.font = UIFont(name:"Avenir", size:16)
+            
+            if (isShowingFacilities) {
+                if (indexPath.row <= facilities.count-1) {
```

Comment:
Please remove this check and the other one for the number of sites. If there are more rows than actual facilities/sites, that is a fatal design flaw that needs to be corrected, not recovered from.